lib/unleash_proxy_client_flutter.dart

```diff
@@ -355,7 +353,12 @@ class UnleashClient extends EventEmitter {
 
   Future<void> setContextFields(Map<String, String> fields) async {
     if (!_anyFieldHasChanged(fields)) return;
-    if (clientState == ClientState.ready) {
+    if (started == false) {
+      await _waitForEvent('initialized');
```

Comment:
Will clientstate be ready after waiting for the initialized event? Just curious as to why we're reusing the exact same lines in both conditions here. But I'm not entirely sure what all the states are, and how "started" relates to "clientState", so 🤷🏼 

As an outsider, it kinda reads like "if we haven't started, then wait for initialized", but I'd assume that if we do that, then the client state will be ready, so I'm not sure what that next if check is for? 

Probably not important to the PR, though.